src/scenes/AcceleratorRouter/Deliverables/QuestionsDeliverableView.tsx

```diff
@@ -32,141 +49,312 @@ export type Props = EditProps & {
 
 const QuestionBox = ({
   item,
-  optionsMenu,
   projectId,
   index,
+  reload,
 }: {
   item: VariableWithValues;
-  optionsMenu: React.ReactNode;
   projectId: number;
   index: number;
+  reload: () => void;
 }): JSX.Element => {
   const theme = useTheme();
 
   const [showRejectDialog, setShowRejectDialog] = useState<boolean>(false);
   const [editing, setEditing] = useState(false);
+  const [values, setValues] = useState<VariableValueValue[]>([]);
+  const dispatch = useAppDispatch();
+  const [requestId, setRequestId] = useState('');
+  const { activeLocale } = useLocalization();
+  const updateResponse = useAppSelector(selectUpdateVariableWorkflowDetails(requestId));
+  const [showEditFeedbackModal, setShowEditFeedbackModal] = useState(false);
+  const [modalFeedback, setModalFeedback] = useState(
+    item.variableValues && item.variableValues.length > 0 && item.variableValues[0].feedback
+      ? item.variableValues[0].feedback
+      : ''
+  );
+
+  useEffect(() => {
+    if (updateResponse?.status === 'success') {
+      reload();
+    }
+  }, [updateResponse]);
+
+  const setStatus = (status: VariableStatusType, feedback?: string, internalComment?: string) => {
+    const request = dispatch(
+      requestUpdateVariableWorkflowDetails({ status, feedback, internalComment, projectId, variableId: item.id })
+    );
+    setRequestId(request.requestId);
+  };
 
-  const rejectItem = () => {
-    return true;
+  const rejectItem = (feedback: string) => {
+    setStatus('Rejected', feedback);
   };
 
   const approveItem = () => {
-    return true;
+    setStatus('Approved');
   };
 
   const onEditItem = () => {
     setEditing(true);
   };
 
-  return (
-    <Box key={`question-${index}`}>
-      {showRejectDialog && <RejectDialog onClose={() => setShowRejectDialog(false)} onSubmit={rejectItem} />}
-      <Box
-        sx={{
-          marginBottom: theme.spacing(4),
-          '&:hover': {
-            background: theme.palette.TwClrBgHover,
-            '.actions': {
-              display: 'block',
+  const onUpdateInternalComment = (internalComment: string) => {
+    const currentStatus: VariableStatusType =
+      item.variableValues && item.variableValues.length > 0 && item.variableValues[0].status
+        ? item.variableValues[0].status || 'Not Submitted'
+        : 'Not Submitted';
+    setStatus(currentStatus, undefined, internalComment);
+  };
+
+  const onValuesChanged = (variableId: number, values: VariableValueValue[]) => {
+    setValues(values);
+  };
+
+  const onSave = async () => {
+    const operations: components['schemas']['AppendValueOperationPayload'][] = [];
+    values.forEach((val) => {
```

Comment:
I wonder if you also need to implement "update". This was in the code I copied over from the document producer variable update.

https://github.com/terraware/terraware-web/pull/2808/files#diff-4de89dc1f5fbcbba072393e8890f56c09bc4cbc145f052cf8e25400efc68ae6bR143

I will check to see what the difference is in the back end

Reply:
I was thinking about that. Maybe if the variable can have multiple values I'm just adding a new one instead of replacing the old one. But for the two I was testing with, looks like they can have only one value (project size and

Reply:
The different operation descriptions are in here: https://github.com/terraware/terraware-server/blob/main/src/main/kotlin/com/terraformation/backend/documentproducer/api/ValueOperationPayloads.kt#L58

Reply:
It seems like what you have will work for simple variable types like single line text, single select dropdown, number, date, etc... but it might not work when dealing with "list" type variables like multi line text, multi select dropdown, etc...

Reply:
So I think we should probably re-use the existing mechanism, since it was already tested on the document producer side and probably works how we need for all the variable types. The rest of this PR looks good, so I am OK with you merging this in as-is and I can create a custom hook or something that will encapsulate that functionality and make sure it is the same across participant and accelerator sides. The "workflow details" stuff doesn't need to change at all.

Reply:
Ok, sounds good! Will fix conflicts and merge